warehouse/transformer/transformer.go

```diff
@@ -71,22 +72,47 @@ func (t *Transformer) Transform(_ context.Context, clientEvents []types.Transfor
 		t.statsFactory.NewTaggedStat("warehouse_dest_transform_output_failed_events", stats.HistogramType, tags).Observe(float64(len(res.FailedEvents)))
 	}()
 
-	for _, clientEvent := range clientEvents {
-		r, err := t.processWarehouseMessage(c, &clientEvent)
-		if err != nil {
-			res.FailedEvents = append(res.FailedEvents, transformerResponseFromErr(&clientEvent, err))
-			continue
+	results := make(chan types.TransformerResponse, len(clientEvents))
+	done := make(chan struct{})
+
+	g := errgroup.Group{}
+	g.SetLimit(t.config.concurrentTransformations.Load())
+
+	go func() {
+		defer close(done)
+		for r := range results {
+			if r.Error != "" {
+				res.FailedEvents = append(res.FailedEvents, r)
+			} else {
+				res.Events = append(res.Events, r)
```

Comment:
I am assuming that order of events in `res` won't be same as the order in `clientEvents`. Is that okay ?

Reply:
Order of events don't matter.